### PR TITLE
Stats package: Add XMLRPC endpoint

### PR DESCRIPTION
**Pull request by fgiannar**
Spin off from: #26348 

#### Changes proposed in this Pull Request:

* Stats related XMLRPC endpoints are now handled by `Automattic\Jetpack\Stats\XMLRPC`

#### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

p9dueE-5Rd-p2

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:
- Look at the code and unit tests

### CODE REVIEW

File: projects/packages/stats/src/class-xmlrpc.php [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+			'page_on_front'       => get_option( 'page_on_front' ),
+			'permalink_structure' => get_option( 'permalink_structure' ),
+			'category_base'       => get_option( 'category_base' ),
+			'tag_base'            => get_option( 'tag_base' ),
+		);
+		$blog = array_merge( Options::get_options(), $blog );
+		unset( $blog['roles'], $blog['blog_id'] );
+
+		add_filter( 'esc_html', array( $this, 'filter_esc_html_check_if_string' ), 10, 2 );
+		$blog = map_deep( $blog, 'esc_html' );
```
Comment (fgiannar) [inline projects/packages/stats/src/class-xmlrpc.php:90]: I'd like to explain this a bit.
Looking at the corresponding [Stats module code](https://github.com/Automattic/jetpack/blob/d38405b569aad92c4164b61acb57b884dc11b2c0/projects/plugins/jetpack/modules/stats.php#L1045) there is the `stats_esc_html_deep` function which I can confirm is not used anywhere outside of that file.
I'd really like to deprecate that function, as most of its functionality (especially the recursive logic) is already covered by `map_deep` core function.
I had to add the filter to make sure boolean values retuned by `get_blog` are not converted to strings, therefore ensure backwards compatibility.

**Comment by sergeymitr** [comment]
Hi @fgiannar.
Since we're adding the new `jetpack.getBlog` endpoint handler, should we remove [the old one](https://github.com/Automattic/jetpack/blob/aa22229b84e3170827a4f067bc9650fea789a9cb/projects/plugins/jetpack/modules/stats.php#L1069)?

**Comment by fgiannar** [comment]
> Hi @fgiannar. Since we're adding the new `jetpack.getBlog` endpoint handler, should we remove [the old one](https://github.com/Automattic/jetpack/blob/aa22229b84e3170827a4f067bc9650fea789a9cb/projects/plugins/jetpack/modules/stats.php#L1069)?

Hey @sergeymitr, thanks for reviewing! Actually the stats package is not yet added to the Jetpack plugin or any other package/plugin therefore any introduced functionality is not affecting anything so far.
This why the testing instructions are limited to the code and unit tests.

File: projects/packages/stats/src/class-xmlrpc.php (hunk 1 above)
Comment (darssen) [reply projects/packages/stats/src/class-xmlrpc.php:90]: Yes, it looks cleaner this way. :)

**Review by darssen — APPROVED** [review]
Things look good after checking the code and unit tests. I also run the unit tests and they work fine.

File: projects/packages/stats/src/class-xmlrpc.php [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+		return $methods;
+	}
+
+	/**
+	 * Stats Get Blog.
+	 *
+	 * @return array
+	 */
+	public function get_blog() {
```
Comment (brbrr) [inline projects/packages/stats/src/class-xmlrpc.php:69]: It is kinda random, and not sure if it makes sense. 

Assuming every `get_option` call will query the DB, and there is no caching added, I wonder if this method could lead to some performance issues. Also, it seems that the returned value wouldn't change between calls.

**Review by adimoldovan — APPROVED** [review]
Looks good!

File: projects/packages/stats/src/class-xmlrpc.php [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+use Automattic\Jetpack\Constants;
+
+/**
+ * Stats XMLRPC.
+ *
+ * Adds additional methods to the WordPress XML-RPC API for handling Stats specific features.
+ *
+ * @since $$next-version$$
+ */
+class XMLRPC {
```
Comment (brbrr) [inline projects/packages/stats/src/class-xmlrpc.php:19]: Does the class name explain its purpose well? For me, it doesn't :) Something like `XMLRPCProvider` might be more self explanatory.

**Review by brbrr — COMMENTED** [review]
Kinda late review :)

File: projects/packages/stats/src/class-xmlrpc.php (hunk 2 above)
Comment (fgiannar) [reply projects/packages/stats/src/class-xmlrpc.php:69]: Good point. TBH at this point I'm trying to slowly refactor the [stats module code](https://github.com/Automattic/jetpack/blob/d38405b569aad92c4164b61acb57b884dc11b2c0/projects/plugins/jetpack/modules/stats.php) by bringing basic pieces of functionality to the Stats package.
There are lots of things that I'd like to change, however trying to draw a reasonable line and find a balance between refactoring, ensuring backwards compatibility and making quick progress.
All this to prepare you for future reviews :)
For now, I hope it's ok to add a todo here to add caching in a follow up PR.
In general, I've only found this endpoint being used by the Jetpack Debugger so I wouldn't be too concerned about performance issues under this context.
Many thanks for taking a look!

File: projects/packages/stats/src/class-xmlrpc.php (hunk 3 above)
Comment (fgiannar) [reply projects/packages/stats/src/class-xmlrpc.php:19]: I chose this name mostly to be consistent with [similar implementations in other packages](https://github.com/Automattic/jetpack/blob/trunk/projects/packages/videopress/src/class-xmlrpc.php).
I'm fine changing it though as I agree with you :)

**Pull request merged**